indexer/services/comlink/src/controllers/api/v4/historical-funding-payment-controller.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  BlockFromDatabase,
+  BlockTable,
+  DEFAULT_POSTGRES_OPTIONS,
+  FundingIndexMap,
+  PerpetualMarketFromDatabase,
+  PerpetualMarketTable,
+  PerpetualPositionFromDatabase,
+  PerpetualPositionStatus,
+  PerpetualPositionTable,
+  SubaccountFromDatabase,
+  SubaccountTable,
+} from '@dydxprotocol-indexer/postgres';
+import express from 'express';
+import { matchedData } from 'express-validator';
+import {
+  Controller, Get, Route, Path, Query,
+} from 'tsoa';
+
+import { getReqRateLimiter } from '../../../caches/rate-limiters';
+import { NotFoundError } from '../../../lib/errors';
+import {
+  getFundingIndexMaps,
+  getPerpetualPositionsWithUpdatedFunding,
+  handleControllerError,
+  initializePerpetualPositionsWithFunding,
+} from '../../../lib/helpers';
+import { rateLimiterMiddleware } from '../../../lib/rate-limit';
+import {
+  CheckSubaccountSchema,
+  CheckTickerParamSchema,
+} from '../../../lib/validation/schemas';
+import { handleValidationErrors } from '../../../request-helpers/error-handler';
+import {
+  HistoricalFundingRequest,
+  HistoricalFundingPaymentResponse,
+  MarketType,
+  PerpetualPositionWithFunding,
+} from '../../../types';
+
+const router: express.Router = express.Router();
+
+@Route('historicalFundingPayment')
+class HistoricalFundingPaymentController extends Controller {
+  @Get('/:ticker')
+  async getHistoricalFundingPayment(
+    @Path() ticker: string,
+      @Query() address: string,
+      @Query() subaccountNumber: number,
+      //   @Query() limit?: number,
+      //   @Query() effectiveBeforeOrAtHeight?: number,
```

Comment:
Do you think it makes sense to include a parameter to indicate whether or not to include unsettledFunding in the response? I think computing unsettledFunding is a lot of additional db calls that might be unnecessary for some API use case.